.github/workflows/tests.yml

```diff
@@ -8,7 +8,12 @@ on:
 jobs:
   python-tests:
     name: python
-    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+        python-version: ['3.9', '3.10', '3.11', '3.12']
```

Comment:
I don't intend to keep all of these here (well, maybe if they all happen to pass :sweat_smile:), but am curious what will and what will not work.